plugins/hls-overloaded-record-dot-plugin/src/Ide/Plugin/OverloadedRecordDot.hs

```diff
@@ -135,56 +154,83 @@ instance Pretty RecordSelectorExpr where
 instance NFData RecordSelectorExpr where
   rnf = rwhnf
 
+-- |The data that is serialized and placed in the data field of resolvable
+-- code actions
+data ORDResolveData = ORDRD {
+  -- |We need the uri to get shake results
+  uri      :: Uri
+  -- |The unique id that allows us to find the specific codeAction we want
+, uniqueID :: Int
+} deriving (Generic, Show)
+instance ToJSON ORDResolveData
+instance FromJSON ORDResolveData
+
 descriptor :: Recorder (WithPriority Log) -> PluginId
                 -> PluginDescriptor IdeState
 descriptor recorder plId = (defaultPluginDescriptor plId)
     { pluginHandlers =
-        mkPluginHandler SMethod_TextDocumentCodeAction codeActionProvider
+      mkCodeActionHandlerWithResolve codeActionProvider resolveProvider
     , pluginRules = collectRecSelsRule recorder
     }
 
+resolveProvider :: PluginMethodHandler IdeState 'Method_CodeActionResolve
+resolveProvider ideState pId ca@(CodeAction _ _ _ _ _ _ _ (Just resData)) =
+    pluginResponse $ do
+        case fromJSON resData of
+            Success (ORDRD uri int) -> do
+                nfp <- getNormalizedFilePath uri
+                CRSR _ crsDetails exts <- collectRecSelResult ideState nfp
+                pragma <- getFirstPragma pId ideState nfp
+                case IntMap.lookup int crsDetails of
+                    Just rse -> pure $ ca {_edit = mkWorkspaceEdit uri rse exts pragma}
+                    -- We need to throw a content modified error here, but we need fendor's
+                    -- plugin error response pr to make it convenient to use here.
+                    _        -> throwE "Content Modified Error"
```

Comment:
Looking at the spec, I'm not sure this is the correct meaning of `ContentModified`. I don't think there's actually anything _wrong_ with the client sending us a document change notification in between asking for code actions and resolving them, even if we can't handle that. Perhaps just `RequestFailed`?

Reply:
It was mentioned in a GitHub issue that this was the way to do it. https://github.com/microsoft/language-server-protocol/issues/1738

Reply:
Okay, let's link to that when we do this!

Reply:
Bonus points: make a PR upstream to clarify in the spec that this is what you should do!